### PR TITLE
Use TDX quote provider to attest and verify.

### PR DESCRIPTION
**Pull request by vbalain**
go-tpm-tools uses go-tdx-guest's device driver to attest and verify quotes but with the recently upstreamed Unified ABI patch from Intel fetches attestation quotes via ConfigFS subsystem.  go-tdx-guest has incorporated this change and now fetches quote via go-configfs-tsm library instead of device driver's IOCTLs.  Added TDX quote provider in attest module which would be used instead of TDX device driver, and updated unit tests.

### CODE REVIEW

File: client/attest.go [hunk 1]
```diff
@@ -19,9 +20,9 @@ const (
 )
 
 // TEEDevice is an interface to add an attestation report from a TEE technology's
-// attestation driver.
+// attestation driver or quote provider.
 type TEEDevice interface {
```
Comment (vbalain) [inline client/attest.go:24]: Ideally this interface should be renamed to TEEQuoteProvider but not updating it to keep changes to a minimum.

File: client/attest.go [hunk 2]
```diff
@@ -144,10 +145,17 @@ type SevSnpDevice struct {
 
 // TdxDevice encapsulates the TDX attestation device to add its attestation quote
 // to a pb.Attestation.
+// Deprecated: TdxDevice is deprecated. It is recommended to use TdxQuoteProvider.
```
Comment (vbalain) [inline client/attest.go:148]: Deprecated!

File: client/attest.go [hunk 3]
```diff
@@ -234,6 +238,73 @@ func (d *TdxDevice) Close() error {
 	return nil
 }
 
+// CreateTdxQuoteProvider creates the TDX quote provider and wraps it with behavior
+// that allows it to add an attestation quote to pb.Attestation.
+func CreateTdxQuoteProvider() (*TdxQuoteProvider, error) {
```
Comment (vbalain) [inline client/attest.go:243]: TDX QuoteProvider uses same interface as a TEEDevice.

File: go.mod [hunk 4]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/google/go-attestation v0.5.0
 	github.com/google/go-cmp v0.5.9
 	github.com/google/go-sev-guest v0.9.3
-	github.com/google/go-tdx-guest v0.2.3-0.20231011100059-4cf02bed9d33
+	github.com/google/go-tdx-guest v0.2.3-0.20231222042644-aeefcb0d0eb3
```
Comment (vbalain) [inline go.mod:9]: go-tdx-guest version update to use latest code

File: server/verify_tdx.go [hunk 5]
```diff
@@ -19,7 +18,7 @@ func TdxDefaultOptions() *VerifyTdxOpts {
 }
 
 // VerifyTdxAttestation checks that the TDX attestation quote is valid
-func VerifyTdxAttestation(attestation *tpb.QuoteV4, opts *VerifyTdxOpts) error {
+func VerifyTdxAttestation(attestation any, opts *VerifyTdxOpts) error {
```
Comment (vbalain) [inline server/verify_tdx.go:21]: Making API future compatible with other quote formats.

Comment (jkl73) [reply server/verify_tdx.go:21]: This (any) can be confused for people to call this function, could you add more description of "attestation" and potentially how to generate an "attestation"

File: cmd/attest_test.go
Comment (jkl73) [inline cmd/attest_test.go:1]: Seems like TestHardwareAttestationPass is failing in CI, probably just need to update the wanterr

File: server/verify_tdx.go (hunk 5 above)
Comment (vbalain) [reply server/verify_tdx.go]: Added description.

File: cmd/attest_test.go
Comment (vbalain) [reply cmd/attest_test.go:1]: Yes. Command **go test ./...** didn't run tests in go-tpm-tools/cmd directory. Updated expected errors in attest_test.go and verify_test.go in cmd. It should fix CI errors.

**Review by jkl73 — APPROVED** [review]

**Pull request merged**